org.eclipse.scanning.malcolm.core/src/org/eclipse/scanning/malcolm/core/MalcolmDevice.java

```diff
@@ -291,19 +294,16 @@ private void handleConnectionStateChange(boolean connected) {
 			setAlive(connected);
 			if (connected) {
 				logger.info("Malcolm Device '" + getName() + "' connection state changed to connected");
-			    java.awt.EventQueue.invokeLater(new Runnable() {
-			        @Override
-					public void run() {
-						try {
-							if (!succesfullyInitialised) {
-								initialize();
-							} else {
-								getDeviceState();
-							}
-						} catch (MalcolmDeviceException ex) {
-							logger.warn("Unable to initialise/getDeviceState for device '{}' on reconnection", getName(), ex);
+				java.awt.EventQueue.invokeLater(() -> {
+					try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)